src/test/objectstore/store_test.cc

```diff
@@ -3567,11 +3567,8 @@ class SyntheticWorkloadState {
     ++in_flight;
     in_flight_objects.insert(old_obj);
 
-    // *copy* the data buffer, since we may modify it later.
     contents[new_obj].attrs = contents[old_obj].attrs;
-    contents[new_obj].data.clear();
-    contents[new_obj].data.append(contents[old_obj].data.c_str(),
-				  contents[old_obj].data.length());
+    contents[new_obj].data = contents[old_obj].data;
```

Comment:
Unfortunately it doesn't. Just managed to reproduce the issue with this commit cherry-picked...